src/android/CompassListener.java

```diff
@@ -38,259 +40,306 @@ Licensed to the Apache Software Foundation (ASF) under one
 import android.os.Handler;
 import android.os.Looper;
 
+import android.util.Log; // MG
```

Comment:
This can be replaced with Apache Cordova's Logger.

```suggestion
import org.apache.cordova.LOG;
```

Also create a log tag for the `CompassListener` class. E.g.:

```java
private static final String LOG_TAG = "CompassListener";
```

Usage example after making those changes:

```java
LOG.d(LOG_TAG, "Some message here");
```

Since you also removed/commented out the only log line

```
// Log.v("MGMG", "heading: " + heading);
```

Technically you can remove the commented out line and import completely since your not logging anything at all..